src/main/java/edu/ucsb/courses/controllers/ScheduleController.java

```diff
@@ -0,0 +1,129 @@
+package edu.ucsb.courses.controllers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.validation.Valid;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.courses.advice.AuthControllerAdvice;
+import edu.ucsb.courses.entities.AppUser;
+import edu.ucsb.courses.entities.Schedule;
+import edu.ucsb.courses.services.UCSBCurriculumService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+import edu.ucsb.courses.documents.Course;
+import edu.ucsb.courses.documents.CoursePage;
+import edu.ucsb.courses.repositories.ScheduleRepository;
+
+
+@RestController
+@RequestMapping("/api/public")
```

Comment:
Please change this from `/api/public` to `/api/member`

The reason is that `/api/public` is reserved for routes that do not require the user to be logged in and authenticated.   The schedule features require a user to be logged in and authenticated so that we can associate a schedule with a given user.